src/audio/drc/drc_math_generic.c

```diff
@@ -114,6 +114,7 @@ inline int32_t drc_sin_fixed(int32_t x)
 
 	/* input range of sin_fixed() is non-negative */
 	int32_t abs_sin_val = sin_fixed(q_mult(ABS(x), PI_OVER_TWO, 30, 30, 28));
+	abs_sin_val = sat_int32(Q_SHIFT_LEFT((int64_t)abs_sin_val, 30, 31));
```

Comment:
function sin_fixed return value is Q2.30 and for compatible support ,   please see 
conversation link https://github.com/thesofproject/sof/pull/3990#discussion_r607344880

Reply:
we should do this in the function itself though rather than hunting down all the places this is used, we can't fix inflight PRs that have not landed, so better to fix ourselves until we have a plan to fix the output.

Reply:
@cujomalainey , made the  change within the function itself i.e. from Q2.30 to Q1.30